File: cpp/.18 july/1.cpp
#include<iostream>
using namespace std;

class Foo
{
public:
Foo()
{
cout<<"This is Constructor"<<endl;
}
void m()
{
cout<<"This is method"<<endl;
}
~Foo()
{
cout<<"This is destructor"<<endl;
}
};

int main()
{
Foo f;
f.m();

}

File: java core/dec-7-2020/1.java
/*
.non-static method's reference
.Method ref. only can apply in functional interface
.For achieveing method ref we can only give ref to abstract method of interface from another class's method of same type and same paramter. 
*/


interface I
{
void m();
}

class Foo 
{
void met()
{
System.out.println("Hello method reference");
}
}

class Test
{
public static void main(String arg[])
{
Foo f=new Foo();

I i=f::met;

i.m();

}
}